grpc/oob/machine/machine.go

```diff
@@ -116,16 +133,20 @@ func (m Action) BootDeviceSet(ctx context.Context, device string, persistent, ef
 
 	host, user, password, parseErr := m.ParseAuth(m.BootDeviceRequest.Authn)
 	if parseErr != nil {
+		span.SetStatus(codes.Error, "error parsing credentials: "+parseErr.Error())
 		return result, parseErr
 	}
+	span.SetAttributes(attribute.String("bmc.host", host), attribute.String("bmc.username", user))
 	base := "setting boot device: " + m.BootDeviceRequest.GetBootDevice().String()
 	msg := "working on " + base
 	m.SendStatusMessage(msg)
+
 	client := bmclib.NewClient(host, "623", user, password, bmclib.WithLogger(m.Log))
 
 	m.SendStatusMessage("connecting to BMC")
 	err = client.Open(ctx)
```

Comment:
After this call to Open, the client metadata would be very helpful to annotate the span.